File: Server/src/main/java/com/codingfairy/bl/service/NewOldService.java
package com.codingfairy.bl.service;

import com.codingfairy.bl.vo.NewOldVo;
import com.codingfairy.exception.ParamException;
import com.codingfairy.utils.enums.QueryThreshold;
import com.codingfairy.web.json.Tuple;
import com.codingfairy.web.json.analysis.element.NewOldCustomElement;

import java.util.List;

/**
 * Created by cuihao on 2017-05-16.
 * new old customer service
 */
public interface NewOldService {

    List<NewOldVo> getNewOldStat(int startHour, QueryThreshold threshold, int startRange, String probeId) throws ParamException;

    List<Tuple<String,Number>> findByHourAndProbe(int hour, String probeId);

    NewOldVo findById(int id);

    NewOldVo save(NewOldCustomElement newOldCustomElement);
}

File: StatDisplay/src/services/newOldService.js
/**
 * Created by yyy on 2017/5/27.
 */
import {request} from "../utils/request";

export async function getNewOld(params) {
  return request(`/newOld/stat`,{
    method: `POST`,
    body: JSON.stringify({
      probeId: params.probeId,
      startHour: params.startHour,
      startRange: params.startRange,
      threshold: params.threshold
    })
  })
}

export async function getNewOldDetail(params) {
  return request(`/newOld/detail`,{
    method: `POST`,
    body: JSON.stringify({
      hour: params.hour,
      probeId: params.probeId
    })
  })
}
